move-vm-backend/src/warehouse.rs

```diff
@@ -62,45 +66,69 @@ impl AccountData {
 }
 
 /// Move VM storage implementation for Substrate storage.
-pub(crate) struct Warehouse<S: Storage> {
+pub(crate) struct Warehouse<S: Storage, Api: SubstrateAPI> {
     /// Substrate storage implementing the Storage trait
     storage: S,
+    substrate_api: Api,
 }
 
-impl<S: Storage> Warehouse<S> {
-    pub(crate) fn new(storage: S) -> Warehouse<S> {
-        Self { storage }
+impl<S: Storage, Api: SubstrateAPI> Warehouse<S, Api> {
+    pub(crate) fn new(storage: S, substrate_api: Api) -> Warehouse<S, Api> {
+        Self {
+            storage,
+            substrate_api,
+        }
     }
 
     pub(crate) fn apply_changes(&self, changeset: ChangeSet) -> Result<()> {
         for (account, changeset) in changeset.into_inner() {
             let key = account.as_slice();
-            let mut account = match self.storage.get(key) {
+            let mut store_account = match self.storage.get(key) {
                 Some(value) => bcs::from_bytes(&value).map_err(Error::msg)?,
                 _ => AccountData::default(),
             };
 
             let (modules, resources) = changeset.into_inner();
-            AccountData::apply_changes(&mut account.modules, modules)?;
-            AccountData::apply_changes(&mut account.resources, resources)?;
+            // process Deposit
+            for res in resources
+                .iter()
+                .filter(|r| r.0.eq(&DEPOSIT_TEMPLATE))
+                .map(|v| v.1)
+                .collect::<Vec<&Op<Vec<u8>>>>()
+            {
+                match res {
+                    Op::New(data) | Op::Modify(data) => {
+                        if let Ok(deposit) = bcs::from_bytes::<Deposit>(data) {
+                            let (destination, amount) = deposit.into();
+                            // make actual transaction using SubstrateApi
+                            self.substrate_api
+                                .transfer(account, destination, amount)
+                                .map_err(|e| anyhow::Error::msg(e.to_string()))?;
```

Comment:
We should include:
```
use crate::alloc::string::ToString;
```
to use `to_string()`.

Reply:
Done.